src/main/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImpl.java

```diff
@@ -253,9 +253,8 @@ public AcspMembersData createAcspMembersData(final AcspMembersSpec spec) throws
         var acspProfileSpec = new AcspProfileSpec();
         if (spec.getAcspProfile() != null) {
             acspProfileSpec.setStatus(spec.getAcspProfile().getStatus());
-        }
-        if (spec.getAcspProfile() != null) {
             acspProfileSpec.setType(spec.getAcspProfile().getType());
+            acspProfileSpec.setAmlDetails(spec.getAcspProfile().getAmlDetails());
```

Comment:
I wonder if we really need this if statement setting all the fields one by one. Couldn't we just use the spec coming as a parameter?

```
        var acspProfileSpec = new AcspProfileSpec();
        if (spec.getAcspProfile() != null) {
            acspProfileSpec = spec.getAcspProfile();
        }
```
This will also simplify the way we make the assertions in the unit test